### DISCUSSION TITLE
Save Material Slots To Imported Avatars/Objects During Import

### DISCUSSION THREAD

**Discussion by EnergyBun** [Yellow-Dog-Man/Resonite-Issues#3349] [New Features]
When an avatar is brought into Resonite, the materials retain the names given to them from Blender, however, if the users do not manually move the materials to the avatar from it's assets folder before initial saving, the names are lost, which makes it harder to modify later.

What should happen instead is that during the import process for avatars and objects, a Materials slot should be created where all of the named materials are placed automatically.  I discovered last night that this further adds to ease of mass converting the materials.

**Comment by Frooxius** [comment]
You can do this by checking "Place Assets On Object" during import in the advanced settings.

However the cost of this is that it prevents certain optimizations (like de-duplication) from occurring, so you pay some potential performance penalty as a result - this is the reason why we don't do this by default.

**Reply by EnergyBun** [reply]
What is de-duplication exactly?

**Reply by Frooxius** [reply]
It looks for any shared assets/materials in the scene and combines them into a single one where possible. This helps reduce resource usage, especially as the number of items grows.
